src/sif/pedestriancost.cc

```diff
@@ -719,6 +722,9 @@ Cost PedestrianCost::EdgeCost(const baldr::DirectedEdge* edge,
   } else if (edge->roundabout()) {
     factor *= kRoundaboutFactor;
   }
+  if (!edge->lit()) {
+    factor *= unlit_factor_;
+  }
```

Comment:
this may be a preoptimization however, it is possible to do this without a branch (`if`), you can cast the bool to float and multiply with it instead which may help performance somewhat, it certainly wont be slower than the branch statement.
```suggestion
  factor *= (!edge->lit()) * unlit_factor_;
```

Reply:
this is bullcrap sorry, didnt think about hard enough, when it is lit we need the factor to still be 1, so we need some slightly more complicated math to avoid the `if` i'll respond later, you can ignore this for now haha

Reply:
ok this is my actual suggestion to remove the branch:

`factor *= edge->lit() + (!edge->lit() * unlit_factor_);`

Reply:
(maybe just forget about it haha)

Reply:
@chrstnbwnkl actually implemented your suggestion:)